services/ymax-planner/src/watchers/cctp-watcher.ts

```diff
@@ -64,9 +64,11 @@ export const watchCctpTransfer = ({
   timeoutMs = TX_TIMEOUT_MS,
   log = () => {},
   setTimeout = globalThis.setTimeout,
+  signal,
 }: CctpWatch & {
   timeoutMs?: number;
   setTimeout?: typeof globalThis.setTimeout;
+  signal?: AbortSignal;
 }): Promise<boolean> => {
   return new Promise(resolve => {
```

Comment:
Let's follow best practices for dealing with already-aborted signals (cf. https://developer.mozilla.org/en-US/docs/Web/API/AbortSignal#implementing_an_abortable_api ).
```suggestion
  return new Promise(resolve => {
    if (signal?.aborted) {
      resolve(false);
      return;
    }
```